wallet/test/Test/Pos/Wallet/Web/Pending/SubmissionSpec.hs

```diff
@@ -0,0 +1,194 @@
+module Test.Pos.Wallet.Web.Pending.SubmissionSpec
+       ( spec
+       ) where
+
+import           Universum
+
+import           Data.Default (def)
+import           Test.Hspec (Spec, describe)
+import           Test.Hspec.QuickCheck (modifyMaxSuccess)
+import           Test.QuickCheck (Gen, arbitrary, generate)
+import           Test.QuickCheck.Arbitrary.Generic (genericArbitrary)
+import           Test.QuickCheck.Monadic (PropertyM)
+
+import           Pos.Client.Txp.History (TxHistoryEntry (..))
+import           Pos.Core (Timestamp (..), addMicrosecondsToTimestamp)
+import           Pos.Launcher (HasConfigurations)
+import           Pos.Txp.Toil.Failure (ToilVerFailure (..))
+import           Pos.Util.CompileInfo (HasCompileInfo, withCompileInfo)
+import           Pos.Wallet.Web.ClientTypes (CHash (..), CId (..))
+
+import           Pos.Wallet.Web.Pending.Submission (TxSubmissionMode, TxSubmissionResult (..),
+                                                    submitAndSavePtxStates)
+import           Pos.Wallet.Web.Pending.Types (PendingTx (..), PtxCondition (..), PtxPoolInfo)
+import           Pos.Util.QuickCheck.Property (assertProperty)
+import           Test.Pos.Configuration (withDefConfigurations)
+import           Test.Pos.Wallet.Web.Mode (walletPropertySpec)
+
+
+-- stack test cardano-sl-wallet --fast --test-arguments "-m Wallet.Web.Pending.SubmissionSpec"
+spec :: Spec
+spec = withCompileInfo def $
+       withDefConfigurations $
+    describe "Wallet.Web.Pending.SubmissionSpec" $ modifyMaxSuccess (const 100) $ do
+        describe "submitAndSaveTx" $ do
+            describe "Normal application" normalApplicationSpec
+            describe "Tx timeout after an hour " txTimeoutSpec
+            describe "ToilKnown exception occurs " toilKnownLargeSpec
+            describe "ToilTooLarge exception occurs " toilTooLargeSpec
```

Comment:
You get e.g. `Wallet.Web.Pending.SubmissionSpec.Normal application.Normal application: 'PtxApplying -> TxApplying'`, is duplication intentional?

Reply:
I just loved the idea of passing description right in function which contains the test

Reply:
It's not duplication, one is `PtxApplying`, the other is `TxApplying`. `TxApplying` is the one I added (for internal state and testing), while `PtxApplying` is already existing.

Reply:
I mean, `describe "Normal application"` and later `walletPropertySpec "Normal application ..."`. Do you intentionally duplicate it in `describe`?

Reply:
Yes? I don't know, it seems to be following the same convention as others.

Reply:
Lool, okay :bug: